pkg/tls/ca.go

```diff
@@ -167,6 +175,30 @@ func (ca *CA) createTemplate(publicKey *ecdsa.PublicKey) x509.Certificate {
 	}
 }
 
+// PEMEncodeCert returns the PEM encoding of cert.
+func PEMEncodeCert(cert []byte) ([]byte, error) {
```

Comment:
Since this func is only ever called on the `Certificate` field of the `CertificateAndPrivateKey` struct, we could do some information hiding and attach this function directly to that struct. Something like:

```
func (cpk *CertificateAndPrivateKey) EncodedCertificate() ([]byte, error) {
	buf := &bytes.Buffer{}
	err := pem.Encode(buf, &pem.Block{Type: "CERTIFICATE", Bytes: cpk.Certificate})
	return buf.Bytes(), err
}
```

Note that I also recommend renaming it to `EncodedCertificate`, to parallel the naming of the `Certificate` field in the struct.

Reply:
ok that looks good :+1: